src/Fantomas.Core/Context.fs

```diff
@@ -338,6 +338,18 @@ let lastWriteEventIsNewline ctx =
         | _ -> false)
     |> Option.defaultValue false
 
+let lastWriteEventIsDotLambda ctx =
```

Comment:
I'm not a fan of this approach. At some point, you actually know you have a dot lambda.
By looking at the events you are just being lucky.
I would prefer to see this solved at the Oak level

Reply:
Sorry for taking that long to come back to this.
I hope this is better.